src/Nix/String.hs

```diff
@@ -139,3 +140,12 @@ principledMakeNixStringWithSingletonContext s c = NixString s (S.singleton c)
 -- | Create a NixString using a singleton context
 principledMakeNixString :: Text -> S.HashSet StringContext -> NixString
 principledMakeNixString s c = NixString s c
+
+principledContextToReferences :: NixString -> S.HashSet StringContext
```

Comment:
What's this?

Reply:
The start of my implementation of this:

https://github.com/NixOS/nix/blob/master/src/libexpr/primops.cc#L1009

Reply:
Oh, we don't need that, we have explicit `ContextFlavors` instead.

Reply:
Removed.